app/src/main/java/woowacourse/shopping/shopping/ProductsItemType.kt

```diff
@@ -0,0 +1,15 @@
+package woowacourse.shopping.shopping
+
+import woowacourse.shopping.model.ProductUIModel
+
+sealed interface ProductsItemType {
+    companion object {
+        const val TYPE_HEADER = 0
+        const val TYPE_ITEM = 1
+        const val TYPE_FOOTER = 2
+    }
+}
+
+data class RecentProductsItem(val product: List<ProductUIModel>) : ProductsItemType
+data class ProductItem(val product: ProductUIModel) : ProductsItemType
+object ProductReadMore : ProductsItemType
```

Comment:
sealed interface의 특성을 살려 다음과 같이 작성해볼 수도 있을 것 같아요!
```suggestion
sealed interface ProductsItemType {

    val viewType: Int

    data class RecentProductsItem(val product: List<ProductUIModel>) : ProductsItemType {
        override val viewType: Int = ...
    }

    data class ProductItem(val product: ProductUIModel) : ProductsItemType {
        override val viewType: Int = ...
    }

    object ProductReadMore : ProductsItemType {
        override val viewType: Int = ...
    }
}
```

다음 코드를 매우 단순화할 수 있답니다 🙂  (이건 직접 해보기!)
```kotlin
    override fun getItemViewType(position: Int): Int {
        return when (productItemTypes[position]) {
            is RecentProductsItem -> ProductsItemType.TYPE_HEADER
            is ProductItem -> ProductsItemType.TYPE_ITEM
            is ProductReadMore -> ProductsItemType.TYPE_FOOTER
        }
    }
```